--- ./dialogs/keyboard-settings/xfce-keyboard-settings.c.org	2010-04-30 18:29:46.201317308 +0200
+++ ./dialogs/keyboard-settings/xfce-keyboard-settings.c	2010-04-30 18:30:34.551318908 +0200
@@ -453,7 +453,7 @@ xfce_keyboard_settings_finalize (GObject
 
 #ifdef HAVE_LIBXKLAVIER
   /* Stop xklavier engine */
-  xkl_engine_stop_listen (settings->priv->xkl_engine);
+  xkl_engine_stop_listen (settings->priv->xkl_engine,XKLL_TRACK_KEYBOARD_STATE);
 #endif /* HAVE_LIBXKLAVIER */
 
   g_object_unref (settings->priv->provider);
